### PR TITLE
chore: `search` command infra

### PR DESCRIPTION
**Pull request by arendjr**
## Summary

This adds the infrastructure for implementing the `biome search` command, but the actual implementation will be left for a future PR. The command remains hidden through the `bpaf` options as long as implementation is not complete.

## Test Plan

Everything should remain green.

### CODE REVIEW

File: crates/biome_service/src/workspace.rs [hunk 1]
```diff
@@ -116,16 +116,17 @@ impl FileFeaturesResult {
...
     }
 
-    /// By default, all features are not supported by a file.
-    const WORKSPACE_FEATURES: [(FeatureName, SupportKind); 3] = [
+    /// By default, only search is supported.
+    const WORKSPACE_FEATURES: [(FeatureName, SupportKind); 4] = [
         (FeatureName::Lint, SupportKind::FileNotSupported),
         (FeatureName::Format, SupportKind::FileNotSupported),
         (FeatureName::OrganizeImports, SupportKind::FileNotSupported),
+        (FeatureName::Search, SupportKind::Supported),
```
Comment (ematipico) [inline crates/biome_service/src/workspace.rs:124]: The default value **must** be `FileNotSupported`. Biome inspects all files, even the ones that can't handle.

File: crates/biome_cli/src/logging.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ pub fn setup_cli_subscriber(level: LoggingLevel, kind: LoggingKind) {
     };
 }
 
-#[derive(Debug, Default, Clone, Ord, PartialOrd, Eq, PartialEq)]
+#[derive(Copy, Debug, Default, Clone, Ord, PartialOrd, Eq, PartialEq)]
```
Comment (ematipico) [inline crates/biome_cli/src/logging.rs:43]: We can revert the change. Not related to the PR.

File: crates/biome_cli/src/logging.rs [hunk 3]
```diff
@@ -138,7 +138,7 @@ impl<S> Filter<S> for LoggingFilter {
 }
 
 /// The kind of logging
-#[derive(Debug, Default, Clone, Eq, PartialEq)]
+#[derive(Copy, Debug, Default, Clone, Eq, PartialEq)]
```
Comment (ematipico) [inline crates/biome_cli/src/logging.rs:141]: We can revert the change. Not related to the PR.

File: crates/biome_cli/src/execute/traverse.rs [hunk 4]
```diff
@@ -746,6 +750,9 @@ impl<'ctx, 'app> TraversalContext for TraversalOptions<'ctx, 'app> {
             TraversalMode::Lint { .. } => file_features.supports_lint(),
             // Imagine if Biome can't handle its own configuration file...
             TraversalMode::Migrate { .. } => true,
+            // We do need to have a suitable parser, but in principle, every
+            // file is searchable.
```
Comment (ematipico) [inline crates/biome_cli/src/execute/traverse.rs:754]: This isn't true. Biome inspects all files, even the ones that can't handle `.png`, `.rs`... any file extension that can't handle. Having `true` here means that it can **search** any file, which isn't the case. Maybe `false` is a better default for now

Comment (arendjr) [reply crates/biome_cli/src/execute/traverse.rs:754]: How does that differ from the migrate command, which also uses `true` here? Surely it wouldn’t migrate images either. My thinking was that every file can be passed to the search command, and depending on the language of the pattern to match, the command will determine if that file should be parsed at all.

File: crates/biome_cli/src/logging.rs (hunk 2 above)
Comment (arendjr) [reply crates/biome_cli/src/logging.rs:43]: It happened from the small refactoring of `logging_level()` and `logging_kind()` in `commands/mod.rs`. It seemed a small enough DX improvement it wouldn’t require a separate PR 😅

File: crates/biome_cli/src/execute/traverse.rs (hunk 4 above)
Comment (ematipico) [reply crates/biome_cli/src/execute/traverse.rs:754]: > How does that differ from the migrate command, which also uses true here? 

The `migrate` command doesn't do any traversal. It just reads the `biome.json`/`biome.jsonc` file (and few more), but it doesn't read files that can or can't handle. It always know how to handle the files that reads.

> My thinking was that every file can be passed to the search command, and depending on the language of the pattern to match, the command will determine if that file should be parsed at all.

This is the same assumption for the other features, and then the value changes based on the file's extension.

File: crates/biome_cli/src/logging.rs (hunk 2 above)
Comment (ematipico) [reply crates/biome_cli/src/logging.rs:43]: Fair. It wasn't mentioned in the PR description (did I miss it?)

File: crates/biome_cli/src/commands/mod.rs [hunk 5]
```diff
@@ -277,6 +278,40 @@ pub enum BiomeCommand {
         sub_command: Option<MigrateSubCommand>,
     },
 
+    /// Searches for Grit patterns across a project.
+    #[bpaf(command, hide)] // !! Command is hidden until ready for release.
+    Search {
+        /// The GritQL pattern to search for.
+        ///
+        /// Note that the search command (currently) does not support rewrites.
+        #[bpaf(positional("PATH"))]
```
Comment (pacak) [inline crates/biome_cli/src/commands/mod.rs:287]: All positional items must go at the end of the block, after all non-positional items.

File: crates/biome_cli/src/execute/traverse.rs (hunk 4 above)
Comment (arendjr) [reply crates/biome_cli/src/execute/traverse.rs:754]: Gotcha 👍

File: crates/biome_cli/src/logging.rs (hunk 2 above)
Comment (arendjr) [reply crates/biome_cli/src/logging.rs:43]: You’re right, should’ve mentioned it.

File: crates/biome_cli/src/commands/mod.rs (hunk 5 above)
Comment (arendjr) [reply crates/biome_cli/src/commands/mod.rs:287]: Thanks for catching! Seems `check_invariants()` didn't actually catch this, but it's a good thing to have anyway :+1:

Comment (pacak) [reply crates/biome_cli/src/commands/mod.rs]: Hmm... That's interesting, seems like a bug. Do you have a commit with it present and not catching?

File: crates/biome_cli/src/logging.rs (hunk 2 above)
Comment (arendjr) [reply crates/biome_cli/src/logging.rs:43]: Reverting after all, since it creates a lot of clippies elsewhere. Happy to make the change in a separate PR still :)
